tensorflow/python/ops/numpy_ops/np_math_ops.py

```diff
@@ -385,14 +385,14 @@ def kron(a, b):  # pylint: disable=missing-function-docstring
   # pylint: disable=protected-access,g-complex-comprehension
   a, b = np_array_ops._promote_dtype(a, b)
   t_a = np_utils.cond(
-      a.ndim < b.ndim,
+      tf.rank(a) < tf.rank(b),
```

Comment:
Currently we don't really guarantee any usability of `tf.experimental.numpy` symbols without calling `tf.experimental.numpy.experimental_enable_numpy_behavior`, so please still try to call `experimental_enable_numpy_behavior` first.

If you really want to get rid of `.ndim`, replacing it with `.shape.rank` is easier because the two are basically the same. I recommend this route.

`tf.rank` is more complicated, because it returns a tensor. If you want to use it, you need to use `np_utils.tf_rank` instead, and change `<` here to `np_utils.greater` (or add an `np_utils.less`), because we want to partial-evaluate away the `cond` as much as possible (for performance and shape-inference reasons).

Reply:
Yes I agree, you should use `.shape.rank`.